### PR TITLE
Categories product

### PR DESCRIPTION
**Pull request by nandvadera1**

### CODE REVIEW

File: app/Models/Product.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Product extends Model
+{
+    use HasFactory;
+
+    protected $fillable = ['category_id', 'name'];
```
Comment (gadhiamitul41) [inline app/Models/Product.php]: I think we don't need category_id in the $fillable

File: app/Http/Controllers/ProductController.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    public function store(Request $request)
+    {
+        $attributes = $request->validate([
+            'category_id' => 'required',
+            'name' => 'required|max:255',
+        ]);
+
+        Product::create($attributes);
```
Comment (gadhiamitul41) [inline app/Http/Controllers/ProductController.php]: I think this should be 
$category = Category::findOrFail($request->get('category_id'));
$category->products()->create($attributes);

File: app/Models/Product.php (hunk 1 above)
Comment (nandvadera1) [reply app/Models/Product.php]: Done

File: app/Http/Controllers/CategoryController.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        $category->update($attributes);
+
+        return redirect('/admin/categories')->with('success', 'Category Updated Successfully!!');
+    }
+
+    public function destroy(Category $category)
+    {
+        $category->delete();
+
+        return back();
```
Comment (gadhiamitul41) [inline app/Http/Controllers/CategoryController.php]: @nandvadera1 : I think this isn't correct.

File: resources/views/users/index.blade.php [hunk 4]
```diff
@@ -38,22 +35,32 @@
...
-                    setTimeout(function() {
-                        notification.style.display = 'none';
-                    }, 4000);
+@section('js')
+    <script>
+        $(document).ready(function () {
+            $('#table1').on('click', '.btn_delete', function () {
+                var id = $(this).data('id');
+                var url = "/admin/users/" + id;
+                var deleteConfirm = confirm("Are you sure to delete User with id : " + id + "?");
```
Comment (gadhiamitul41) [inline resources/views/users/index.blade.php]: I think the error message should be 
**Are you sure you want to delete this user?**

File: resources/views/users/index.blade.php [hunk 5]
```diff
@@ -38,22 +35,32 @@
...
+                    $.ajax(
+                        {
+                            url: url,
+                            type: 'DELETE',
+                            data: {
+                                "id": id,
+                                "_token": token,
+                            },
+                            success: function (response) {
+                                console.log("Delete operation successful");
```
Comment (gadhiamitul41) [inline resources/views/users/index.blade.php]: Message should be 
**User deleted successfully**

File: .gitignore [hunk 6]
```diff
@@ -13,3 +13,4 @@ npm-debug.log
 yarn-error.log
 /.idea
 /.vscode
+storage/clockwork/
```
Comment (gadhiamitul41) [inline .gitignore]: We don't this line as we are adding storage/clockwork/.gitignore

**Comment by gadhiamitul41** [comment]
Looks good to me. Merge it.

**Pull request merged**